Makefile

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+KATA_RUNTIME_NAME = kata-runtime
+KATA_RUNTIME_PATH = /usr/local/bin/$(KATA_RUNTIME_NAME)
+
+ifeq (,$(KATA_RUNTIME))
+    # no argument specified
+    ifeq (,$(MAKECMDGOALS))
+        fail = true
+    endif
+
+    # remove goals that don't require the variable to be set
+    remaining=$(filter-out clean help,$(MAKECMDGOALS))
+
+    ifneq (,$(remaining))
+        fail = true
+    endif
+
+    ifeq ($(fail),true)
+        $(error "ERROR: KATA_RUNTIME not set - run 'make help'")
+    endif
+endif
+
+ifeq (cc,$(KATA_RUNTIME))
+	RUNTIME_DIR = cc-runtime
+	RUNTIME_NAME = kata-runtime
+endif
+
+ifeq (runv,$(KATA_RUNTIME))
+	RUNTIME_DIR = runv
+	RUNTIME_NAME = runv
+endif
+
+default: build
+
+build:
+ifeq (cc,$(KATA_RUNTIME))
+	make -C $(RUNTIME_DIR) build-kata-system
+endif
+ifeq (runv,$(KATA_RUNTIME))
+	(cd $(RUNTIME_DIR) && [ -e configure ] || ./autogen.sh && ./configure && make)
+endif
+
+install:
+ifeq (cc,$(KATA_RUNTIME))
+	make -C $(RUNTIME_DIR) install-kata-system
+endif
+ifeq (runv,$(KATA_RUNTIME))
+	make -C $(RUNTIME_DIR) install
+	ln -s /usr/local/bin/runv $(KATA_RUNTIME_PATH)
+endif
+
+clean:
+	rm -f $(KATA_RUNTIME_PATH)
```

Comment:
You could run `clean` in the (configured, or both?) runtime dirs as well maybe.

Reply:
Done.